library/core/src/option.rs

```diff
@@ -2146,6 +2146,22 @@ impl<T> SpecOptionPartialEq for crate::ptr::NonNull<T> {
     }
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
+impl SpecOptionPartialEq for cmp::Ordering {
+    #[inline]
+    fn eq(l: &Option<Self>, r: &Option<Self>) -> bool {
+        // SAFETY: `Ordering` is a field-less enum, so while this isn't a public
+        // guarantee, here in `core` we can rely on layout optimizations because
+        // there are hundreds of niches available, and this will stop compiling
+        // if a change ever makes the `Option` take more than a byte.
+        unsafe {
+            let l: i8 = mem::transmute(*l);
```

Comment:
Can this be `l.map_or(2, |x| x as i8)` to avoid the transmute?

Reply:
Good point; I'll give that a shot.

@rustbot author

Reply:
Oh, interesting, `map_or(2)` [works](https://play.rust-lang.org/?version=nightly&mode=release&edition=2021&gist=3f0cc1b4ccc3c8f5f5fc9c20bcda295c), but `map_or(-2)` [doesn't](https://play.rust-lang.org/?version=nightly&mode=release&edition=2021&gist=8d17c36cbc67f21dbf2fbabd0e3cb1b8).

Man I wish `!range` metadata was usable in more places...

Reply:
Done.

@rustbot ready